uniswap_swap_detection/src/agent.spec.ts

```diff
@@ -0,0 +1,244 @@
+  const tokenOut = "0xc2132D05D31c914a87C6611C10748AEb04B58e8F";
+  const fee = 3000;
+  const recipient = "0x08d16B72dad2c52FD94835FF49f51514aFcBfBfC";
+  const deadline = 1718308601;
+  const token0 = "0x765Af38A6e8FDcB1EFEF8a3dd2213EFD3090B00F";
+  const token1 = "0xc2132D05D31c914a87C6611C10748AEb04B58e8F";
+  // const amountIn = ethers.BigNumber.from("847955346422803956304");
+  const amountOutMinimum = 31628133;
+  const sqrtPriceLimitX96 = 0;
+
+  const amount0 = ethers.BigNumber.from("847955346422803956304");
+  const amount1 = 31628133;
+  const sqrtPricex96 = 0;
+  const liquidity = 5;
+  const tick = 10;
+  const sender = "0x74993dD14475b25986B6ed8d12d3a0dFf92248f4";
+  const poolAddress = "0x61D3f523cd7e93d8deF89bb5d5c4eC178f7CfE76";
+
+  function createGetToken0() {
+    return mockProvider.addCallTo(address1, 20, iface, "token0", {
+      inputs: [],
+      outputs: [token0],
+    });
+  }
+  function createGetToken1() {
+    return mockProvider.addCallTo(address1, 20, iface, "token1", {
+      inputs: [],
+      outputs: [token1],
+    });
+  }
+  function createGetfee() {
+    return mockProvider.addCallTo(address1, 20, iface, "fee", {
+      inputs: [],
+      outputs: [fee],
+    });
+  }
+
+  describe("token swap detection handle Transaction", () => {
+    it("returns empty findings with TestTransactionEvent", async () => {
+      txEvent = new TestTransactionEvent();
+      txEvent.setBlock(20);
+      createGetToken0();
+      createGetToken1();
+      createGetfee();
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([]);
+    });
+    it("returns findings", async () => {
+      const address =
+        "0x61D3f523cd7e93d8deF89bb5d5c4eC178f7CfE76".toLowerCase();
+      // txEvent.setBlock(20);
+      createGetToken1();
+      createGetToken0();
+      createGetfee();
+
+      txEvent = new TestTransactionEvent().addEventLog(SWAP_EVENT, address, [
+        sender,
+        recipient,
+        amount0,
+        amount1,
+        sqrtPricex96,
+        liquidity,
+        tick,
+      ]);
+
+      txEvent.setBlock(20);
+      const findings = await handleTransaction(txEvent);
+
+      expect(findings).toStrictEqual([
+        Finding.fromObject({
+          name: "Uniswap-1",
+          description: `A swap between ${token0} and ${token1} on UniswapV3 was detected on this pool ${address}`,
+          alertId: "FORTA-1",
+          severity: FindingSeverity.Low,
+          type: FindingType.Info,
+          protocol: "polygon",
+          metadata: {
+            sender,
+            recipient,
+            tokenIn: token0,
+            tokenOut: token1,
+            amount0: amount0.toString(),
+            amount1: amount1.toString(),
+          },
+        }),
+      ]);
+    });
+
+    it("returns empty findings if not uniswap event", async () => {
+      createGetToken0();
+      createGetToken1();
+      createGetfee();
+      const address =
```

Comment:
```suggestion
      const nonUniswapPoolAddress =
```
Suggestiont o rename variable to be more descriptive and easier to read.